shared/api/instance.go

```diff
@@ -1,8 +1,12 @@
 package api
```

Comment:
Please can you add the new API types as a separate commit to ease backporting. Ta

Reply:
Have done!